src/commands/sites/sites-create-template.js

```diff
@@ -0,0 +1,251 @@
+      type: 'list',
+      name: 'templateName',
+      message: 'Template:',
+      choices: templates.map((template) => ({
+        value: template.slug,
+        name: template.name,
+      })),
+    },
+  ])
+
+  const accounts = await api.listAccountsForUser()
+
+  let { accountSlug } = options
+  if (!accountSlug) {
+    const { accountSlug: accountSlugInput } = await inquirer.prompt([
+      {
+        type: 'list',
+        name: 'accountSlug',
+        message: 'Team:',
+        choices: accounts.map((account) => ({
+          value: account.slug,
+          name: account.name,
+        })),
+      },
+    ])
+    accountSlug = accountSlugInput
+  }
+
+  const { name: nameFlag } = options
+  let user
+  let site
+
+  // Allow the user to reenter site name if selected one isn't available
+  const inputSiteName = async (name) => {
+    let siteSuggestion
+    if (!user) user = await api.getCurrentUser()
+
+    if (!name) {
+      let { slug } = user
+      let suffix = ''
+
+      // If the user doesn't have a slug, we'll compute one. Because `full_name` is not guaranteed to be unique, we
+      // append a short randomly-generated ID to reduce the likelihood of a conflict.
+      if (!slug) {
+        slug = slugify(user.full_name || user.email)
+        suffix = `-${uuidv4().slice(0, SITE_NAME_SUGGESTION_SUFFIX_LENGTH)}`
+      }
+
+      const suggestions = [
+        `super-cool-site-by-${slug}${suffix}`,
+        `the-awesome-${slug}-site${suffix}`,
+        `${slug}-makes-great-sites${suffix}`,
+        `netlify-thinks-${slug}-is-great${suffix}`,
+        `the-great-${slug}-site${suffix}`,
+        `isnt-${slug}-awesome${suffix}`,
+      ]
+      siteSuggestion = sample(suggestions)
+
+      console.log(
+        `Choose a unique site name (e.g. ${siteSuggestion}.netlify.app) or leave it blank for a random name. You can update the site name later.`,
+      )
+      const { name: nameInput } = await inquirer.prompt([
+        {
+          type: 'input',
+          name: 'name',
+          message: 'Site name (optional):',
+          filter: (val) => (val === '' ? undefined : val),
+          validate: (input) => /^[a-zA-Z\d-]+$/.test(input) || 'Only alphanumeric characters and hyphens are allowed',
+        },
+      ])
+      name = nameInput
+    }
+
+    try {
+      const siteName = name ? name.trim() : siteSuggestion
+
+      // Create new repo from template
+      const createGhRepoResp = await fetch(`https://api.github.com/repos/${templateUrl.templateName}/generate`, {
+        method: 'POST',
+        headers: {
+          Authorization: `token ${ghToken.token}`,
+        },
+        body: JSON.stringify({
+          name: siteName,
+        }),
+      })
+      const resp = await createGhRepoResp.json()
+
+      if (resp.errors) {
+        if (resp.errors[0].includes('Name already exists on this account')) {
+          throw new Error('Duplicate repo')
+        }
+      } else {
+        site = await api.createSiteInTeam({
```

Comment:
This will create the site, but does not set up the CI part. Meaning a commit to the repo won't trigger a build on Netlify.
See https://github.com/netlify/cli/blob/40b1cf9b45e78d07817829323320ea8a90c476e4/src/commands/sites/sites-create.js#L127 if we want the behavior to configure GitHub as well